cardano-cli/src/Cardano/CLI/Compatible/Commands.hs

```diff
@@ -0,0 +1,75 @@
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LambdaCase #-}
+
+{-
+This module is concerned with providing backwards compatible cli commands for our internal
+testing needs. The intention is to restrict as much as possible which functionality we maintain backwards
+compatibility for.
+-}
+
+module Cardano.CLI.Compatible.Commands
+  ( AnyCompatibleCommand (..)
+  , CompatibleCommand (..)
+  , pAnyCompatibleCommand
+  , renderAnyCompatibleCommand
+  )
+where
+
+import           Cardano.Api
+
+import           Cardano.CLI.Compatible.Governance
+import           Cardano.CLI.Compatible.Transaction
+import           Cardano.CLI.Environment
+import           Cardano.CLI.Parser
+
+import           Data.Foldable
+import           Data.Text
+import           Options.Applicative
+import qualified Options.Applicative as Opt
+
+data AnyCompatibleCommand where
+  AnyCompatibleCommand :: CompatibleCommand era -> AnyCompatibleCommand
+
+renderAnyCompatibleCommand :: AnyCompatibleCommand -> Text
+renderAnyCompatibleCommand = \case
+  AnyCompatibleCommand cmd -> renderCompatibleCommand cmd
+
+data CompatibleCommand era
+  = CompatibleTransactionCmd (CompatibleTransactionCmds era)
+  | CompatibleGovernanceCmds (CompatibleGovernanceCmds era)
+
+renderCompatibleCommand :: CompatibleCommand era -> Text
+renderCompatibleCommand = \case
+  CompatibleTransactionCmd cmd -> renderCompatibleTransactionCmd cmd
+  CompatibleGovernanceCmds cmd -> renderCompatibleGovernanceCmds cmd
+
+pAnyCompatibleCommand :: EnvCli -> Parser AnyCompatibleCommand
```

Comment:
A parser should go to `cardano-cli/src/Cardano/CLI/Compatible/Options.hs`

Reply:
I don't want to mirror the previous `EraBased` directory structure. A `Common` module is fine but this parser is not common.